src/main/java/org/takes/rq/multipart/RqMultipartBase.java

```diff
@@ -0,0 +1,376 @@
+ *
+ * Copyright (c) 2014-2016 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.rq.multipart;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.net.HttpURLConnection;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.EqualsAndHashCode;
+import org.takes.HttpException;
+import org.takes.Request;
+import org.takes.misc.Sprintf;
+import org.takes.misc.VerboseIterable;
+import org.takes.rq.RqGreedy;
+import org.takes.rq.RqHeaders;
+import org.takes.rq.RqLengthAware;
+import org.takes.rq.RqLive;
+import org.takes.rq.RqMultipart;
+import org.takes.rq.RqWithHeader;
+import org.takes.rq.TempInputStream;
+
+/**
+ * Request decorator, that decodes FORM data from
+ * {@code multipart/form-data} format (RFC 2045).
+ *
+ * <p>For {@code application/x-www-form-urlencoded}
+ * format use {@link org.takes.rq.RqForm}.
+ *
+ * <p>It is highly recommended to use {@link org.takes.rq.RqGreedy}
+ * decorator before passing request to this class.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Yegor Bugayenko (yegor@teamed.io)
+ * @version $Id$
+ * @since 0.32.8
+ * @see <a href="http://www.w3.org/TR/html401/interact/forms.html">
+ *  Forms in HTML</a>
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @see RqGreedy
+ */
+@SuppressWarnings("PMD.ExcessiveImports")
```

Comment:
@mkordas well actually it happens because this Class is indeed highly coupled as raised by checkstyle and most of its dependencies are in the package _org.takes.rq_, so when I moved the class in a different package a lot of import as been added but you are probably right, I should add a puzzle